pkg/alert/alert.go

```diff
@@ -0,0 +1,427 @@
+			stringOrDefault(test.OperatingSystem, "-"),
+			fmt.Sprintf("%d%%", int(test.SuccessRate)),
+			failureReason,
+			test.LastFailedTimestamp,
+		}
+		content = append(content, newRow)
+		row++
+	}
+
+	table.SetHeader(header)
+	table.SetAlignment(tablewriter.ALIGN_CENTER)
+	table.AppendBulk(content)
+	table.Render()
+	return writer.String()
+}
+
+//splitSlackMessage split message line wise based on given characters limit
+func splitSlackMessage(message string, charactersLimit int) []string {
+	var messageSplits []string
+	lines := strings.Split(message, "\n")
+	messageSplitItem := ""
+	for _, line := range lines {
+		if len(messageSplitItem)+len(line) < charactersLimit {
+			messageSplitItem += "\n" + line
+		} else {
+			messageSplits = append(messageSplits, messageSplitItem)
+			messageSplitItem = line
+		}
+	}
+	messageSplits = append(messageSplits, messageSplitItem)
+	return messageSplits
+}
+
+func stringOrDefault(input, defaultValue string) string {
+	if input == "" {
+		return defaultValue
+	}
+	return input
+}
+
+//elasticRequest send HTTP request to elasticsearch
+func (alert *Alert) elasticRequest(urlAttributes, httpMethod, payloadFormated string, result interface{}) {
+	url := fmt.Sprintf("%s%s", alert.cfg.Elasticsearch.Endpoint, urlAttributes)
+	payload := strings.NewReader(payloadFormated)
+	alert.log.Info(fmt.Sprintf("creating HTTP %s request on %s", httpMethod, url))
+	req, err := http.NewRequest(strings.ToUpper(httpMethod), url, payload)
+	if err != nil {
+		alert.log.Error(err, fmt.Sprintf("failed to create a http request for %s", url))
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	req.Header.Add("cache-control", "no-cache")
+	req.Header.Add("Authorization", alert.cfg.Elasticsearch.Authorization)
+
+	res, _ := http.DefaultClient.Do(req)
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		alert.log.Error(err, "failed to read response body %s")
+	}
+	if result != nil {
+		if err := json.Unmarshal(body, result); err != nil {
+			alert.log.Error(err, fmt.Sprintf("failed to unmarshal %s", string(body)))
+		}
+	}
+}
+
+//filePostedAlerts posts test contexts to elasticsearch
+func (alert *Alert) filePostedAlerts(tests map[string]TestDetails) {
+	payload := generatePostedAlertsPayload(tests)
+	alert.elasticRequest("/tm-alert/_doc/_bulk", "POST", payload, nil)
+	alert.log.Info(fmt.Sprintf("filed %d tests as alerted in elasticsearch", len(tests)))
+}
+
+//generatePostedAlertsPayload generates a bulk payload of test context docs
+func generatePostedAlertsPayload(tests map[string]TestDetails) string {
+	datetime := time.Now().UTC().Format("2006-01-02T15:04:05") + "Z"
+	payload := ""
+	for _, test := range tests {
+		payload += fmt.Sprintf(`{ "index":{} }`+"\n"+`{ "testContext":"%s","datetime":"%s" }`+"\n", test.Context, datetime)
+	}
+	return payload
+}
+
+// MessageRequest defines a default slack request for a message
+type MessageRequest struct {
```

Comment:
why do you redefine slack message structures?
they are already defined here: https://github.com/gardener/test-infra/blob/master/pkg/util/slack/slack.go#L96